src/pages/iou/steps/MoneyRequestAmountForm.js

```diff
@@ -70,6 +70,7 @@ function MoneyRequestAmountForm({amount, currency, isEditing, forwardedRef, onCu
 
     const textInput = useRef(null);
 
+    const decimals = useMemo(() => CurrencyUtils.getCurrencyDecimals(currency), [currency]);
```

Comment:
`getCurrencyDecimals` is a very simple function, using `useMemo` probably is more costly than calling the function itself.

<img width="589" alt="image" src="https://github.com/Expensify/App/assets/87341702/8157d5f6-19f7-4262-b341-be035878446b">

If you want to make things faster, maybe storing the regexes in some constants could help? Not sure if we validate on each keystroke or when we submit. 

<img width="859" alt="image" src="https://github.com/Expensify/App/assets/87341702/b3179cc6-8c8c-492d-8661-e6aa557734c9">

Reply:
@aldo-expensify I pushed a commit removing the `useMemo` but I'm not sure of how storing the `regex` in constants  would help.. Are you suggesting we put `^\\d+(,\\d+)*$` in a const like `CONST.VALIDATE_AMOUNT_DIGITS_STRING` for the case when `decimals == 0` but should we keep it the same way for the else part?

Reply:
Gentle bump @aldo-expensify

Reply:
> Are you suggesting we put ^\\d+(,\\d+)*$ in a const like CONST.VALIDATE_AMOUNT_DIGITS_STRING for the case when decimals == 0 but should we keep it the same way for the else part?

Don't worry about this, I think it is best to keep it like it is and only do something if we really see that it slows down anything.